app/lib/schema_file_loader.rb

```diff
@@ -4,13 +4,15 @@ class SchemaFileLoader
   REGION = "us-east-1".freeze
   EFILE_SCHEMAS_FILENAMES = (
     [
-      ["efile1040x_2020v5.1.zip", "irs"],
-      ["efile1040x_2021v5.2.zip", "irs"],
-      ["efile1040x_2022v5.3.zip", "irs"],
-      ["efile1040x_2023v5.0.zip", "irs"]
+      # Format is schema name, directory, and whether the file is optional
+      ["efile1040x_2020v5.1.zip", "irs", false],
+      ["efile1040x_2021v5.2.zip", "irs", false],
+      ["efile1040x_2022v5.3.zip", "irs", false],
+      ["efile1040x_2023v5.0.zip", "irs", false]
     ] +
-      StateFile::StateInformationService.state_schema_file_names.map do |schema_file_name|
-        [schema_file_name, "us_states"]
+      StateFile::StateInformationService::STATES_INFO.map do |key, values|
+        # TODO: If adding another member to this array, consider refactoring to a hash instead
+        [values[:schema_file_name], "us_states", values[:optional]]
```

Comment:
I felt like this was on the very cusp of needing a refactor. I left it alone since this is fully internal to this file just left a note for the future.